doc/stake-pool-operations/KES_period.md

```diff
@@ -1,46 +1,206 @@
+```
+
+**This command increments the cold counter by 1.**
+
+## :warning: Vasil hard fork breaking changes
+
+Note that from the Vasil hard fork onwards there is a new rule for the cold.counter:
+
+A new Operational Certificate will be considered valid only if its counter value is exactly one more than the previous Operational Certificate that has successfully minted at least one block.
+
+Prior to Vasil, it was permitted to use any strictly larger counter value than used on chain previously, but this is no longer permitted. It must be incremented by exactly one. In particular, this means one cannot use the current time or slot number as a counter value.
+
+### Find the counter value used on-chain
+
+To find out the current **Operational Certificate Counter** registered on-chain, run:
+
+```bash
+> cardano-cli query kes-period-info --testnet-magic 42 --op-cert-file node-spo3/opcert.cert
+```
+
+```
+✓ Operational certificate's KES period is within the correct KES period interval
+✓ The operational certificate counter agrees with the node protocol state Operational Certificate
+{
+    "qKesCurrentKesPeriod": 15,
+    "qKesEndKesInterval": 18,
+    "qKesKesKeyExpiry": null,
+    "qKesMaxKESEvolutions": 6,
+    "qKesNodeStateOperationalCertificateNumber": 3,
+    "qKesOnDiskOperationalCertificateNumber": 3,
+    "qKesRemainingSlotsInKesPeriod": 690,
+    "qKesSlotsPerKesPeriod": 300,
+    "qKesStartKesInterval": 12
+}
+```
+
+Where:
+
+* `"qKesNodeStateOperationalCertificateNumber": 3,` is the last counter registered **OnChain** to forge a block.
+
+and
+
+* `"qKesOnDiskOperationalCertificateNumber": 3,` is the counter value of your **OnDisk** operational certificate.
+
+When your pool has already forged a block with the current Operational Certificate the above values will match.
+
+If your pool has **not** yet forged a block with the current Operational Certificate,  `cardano-cli query kes-period-info` shows that `"qKesOnDiskOperationalCertificateNumber"` is greater than `"qKesNodeStateOperationalCertificateNumber"` by exactly one (1), for example:
+
+```json
+"qKesNodeStateOperationalCertificateNumber": 2,
+"qKesOnDiskOperationalCertificateNumber": 3,
+```
+
+Once your pool forges a block `qKesNodeStateOperationalCertificateNumber` will be updated accordingly **(OnChain)** and these values will be the same.
+
+### How do I know if I have issued an _invalid_ Operational Certificate
+
+Run `cardano-cli query kes-period-info` right after you have created the new Operational Certificate. If the **OnDisk** counter differs from the **OnChain** counter by more than 1, your certificate will be _invalid_, for example:
+
+```json
+"qKesNodeStateOperationalCertificateNumber": 2,
+"qKesOnDiskOperationalCertificateNumber": 4,
+```
+
+Any blocks forged with such a certificate will be _invalid blocks_.
+
+Issuing a **valid certificate** after having issued an **invalid** one requires a new **Operational Certificate Counter** and a new Operatinal Certificate.
```

Comment:
Is it worth describing how this can happen?  Will it still possible to make this happen after the hard fork?